modules/cli/src/main/scala/scala/cli/commands/SharedOptions.scala

```diff
@@ -110,7 +110,8 @@ final case class SharedOptions(
   def buildOptions(
     enableJmh: Boolean,
     jmhVersion: Option[String],
-    ignoreErrors: Boolean = false
+    ignoreErrors: Boolean = false,
+    shouldCompileTest: Boolean = false
```

Comment:
Maybe this should be an `Option`, defaulting to `None`, so that nothing is changed in `TestOptions` by default.